### ISSUE TITLE
[Password page] Accessibility issues

### ISSUE THREAD

**Issue by ludoboludo** [Shopify/dawn#2358]
## Describe the current behavior
- lack of `show password` button to confirm the entered password 
- `Login with password` at the top right is too lightly colored and the font is pretty small
- check on the login steps as some users have experienced issues getting a `email invalid` error
- the order of the labels/inputs might be odd. You're hearing first `Password Edit` before you get to the `Password` input. Same thing with the `Email edit` then `Email`. 

Those are feedback received during some usability testing. 

## Version information (Dawn, browsers and operating systems)

- Dawn Version: 8.0.0
- Chrome Version 108.0.5359.124
- macOS Version 13.1
